Code/autopkglib/SignToolVerifier.py

```diff
@@ -0,0 +1,131 @@
+            "description": (
+                "The path to signtool.exe. This varies between versions of the "
+                "Windows SDK, so you can explicitly set that here in an override."
+            ),
+            "default": r"C:\Program Files (x86)\Windows Kits\10\bin\x64\signtool.exe",
+        },
+        "additional_arguments": {
+            "required": False,
+            "description": (
+                "Array of additional argument strings to pass to signtool. "
+                "Note that currently '/v' and '/pa' are always passed."
+            ),
+            "default": None,
+        },
+    }
+    output_variables: Dict[str, Any] = {}
+
+    description: str = __doc__
+
+    def codesign_verify(
+        self,
+        path: str,
+        additional_arguments: Optional[List[str]] = None,
+        signtool_path: Optional[str] = None,
+    ) -> bool:
+        """
+        Runs 'signtool.exe /pa <path>'. Returns True if signtool exited with 0
+        and False otherwise.
+        """
+        if not additional_arguments:
+            additional_arguments = []
+
+        signtool_bin = signtool_path if signtool_path else "signtool.exe"
+        # Call signtool with "/v" to produce information about the signer when run,
+        # and "/pa" to use the "Default Authenticode" Verification Policy.
+        process = [signtool_bin, "verify", "/v", "/pa"] + additional_arguments
+
+        # Makes the path absolute and normalizes it into standard Windows form.
+        # E.g., /Program Files (x86)/Windows Kits/10/bin/x64/signtool.exe will be
+        # converted to the appropriate C:\\... path after this.
+        process.append(os.path.abspath(path))
+
+        # Run signtool with stderr redirected to stdout to ensure that all output
+        # is always captured from the tool.
+        proc = subprocess.Popen(
+            process,
+            stdin=None,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            text=True,
+        )
+        (output, _) = proc.communicate()
+
+        for line in output.replace("\n\n", "\n").replace("\n\n\n", "\n\n").splitlines():
+            self.output(line)
+
+        if proc.returncode == 1:
+            raise ProcessorError(
+                "Authenticode verification failed. Note that all "
+                "verification can be disabled by setting the variable "
+                "DISABLE_CODE_SIGNATURE_VERIFICATION to a non-empty value."
+            )
+        elif proc.returncode == 2:
+            self.output("WARNING: Verification had warnings. Check output above.")
+
+        return proc.returncode == 0
+
+    def main(self):
+        if self.env.get("DISABLE_CODE_SIGNATURE_VERIFICATION"):
+            self.output("Authenticode verification disabled for this recipe run.")
+            return
+
+        input_path = self.env["input_path"]
+        signtool_path = self.env["signtool_path"]
+        additional_arguments = self.env["additional_arguments"]
```

Comment:
Same here.

Reply:
`additional_arguments` has a default value specified as well. Thoughts?

Reply:
I managed to miss that, sorry.